test/eval.cpp

```diff
@@ -155,3 +156,21 @@ TEST(eval, interpret_dynamic_range_sum)
     vector<float> seq{8.0f, 11.0f, 14.0f};
     ASSERT_EQ(result_val, seq);
 }
+
+TEST(eval, interpret_dynamic_non_zero)
+{
+    auto input = make_shared<op::Parameter>(element::f32, PartialShape{});
+    auto non_zero = make_shared<op::v3::NonZero>(input, element::i64);
+    auto fun = make_shared<Function>(OutputVector{non_zero}, ParameterVector{input});
+    auto backend = runtime::Backend::create("INTERPRETER");
+    auto input_val = backend->create_tensor(element::f32, Shape{});
+    copy_data(input_val, vector<float>{1.0f});
+    auto result = make_shared<HostTensor>();
+    auto cfun = backend->compile(fun);
+    cfun->call({result}, {input_val});
```

Comment:
dont we want a test just for evaluate without going through compile? I see both for dynamic_range_sum. Trying to understand why/why not.

Reply:
See the commit with more test cases and also creating op_eval test folder.